encode_test.go

```diff
@@ -243,6 +247,131 @@ func TestTimestampWithOutTimezone(t *testing.T) {
 	test("2013-01-04T20:14:58.80033Z", "2013-01-04 20:14:58.80033")
 }
 
+func TestInfinityTimestamp(t *testing.T) {
+	db := openTestConn(t)
+	defer db.Close()
+	var err error
+	var resultT time.Time
+	var resultI interface{}
+
+	expectedError := fmt.Errorf(`sql: Scan error on column index 0: unsupported driver -> Scan pair: []uint8 -> *time.Time`)
+	type testCases []struct {
+		Query       string
+		Param       string
+		ExpectedErr error
+		ExpectedVal interface{}
+	}
+	tc := testCases{
+		{"SELECT $1::timestamp", "-infinity", expectedError, _infinityString},
+		{"SELECT $1::timestamptz", "-infinity", expectedError, _infinityString},
+		{"SELECT $1::timestamp", "infinity", expectedError, infinityString},
+		{"SELECT $1::timestamptz", "infinity", expectedError, infinityString},
+	}
+	// try to assert []byte to time.Time
+	for _, q := range tc {
+		err = db.QueryRow(q.Query, q.Param).Scan(&resultT)
+		if err.Error() != q.ExpectedErr.Error() {
+			t.Errorf("Scanning -/+infinity, expected error, \"%s\", got \"%s\"", q.ExpectedErr, err)
```

Comment:
It's better to use %q than \"%s\"; less typing and automatically escaped correctly.

Ditto for all similar uses below.